### PR TITLE
feat(integrations): Add integration fields to Knowledge and SyncsUser models

### PR DESCRIPTION
**Pull request by StanGirard**
This pull request adds integration fields to the Knowledge and SyncsUser models. The integration fields include `integration` and `integration_link`. These fields allow for better integration with external systems and provide additional information about the integration.

### CODE REVIEW

File: backend/api/quivr_api/modules/sync/utils/list_files.py [hunk 1]
```diff
@@ -208,31 +216,33 @@ def fetch_files(endpoint, headers):
         )
 
     items = fetch_files(endpoint, headers)
+    logger.debug("!!!!!! The items returned are ", items)
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/list_files.py:219]: remove

File: backend/api/quivr_api/modules/sync/utils/googleutils.py [hunk 2]
```diff
@@ -113,6 +117,7 @@ async def _upload_files(
                     "pptx",
                     "doc",
                 ]:
+                    logger.debug("Getting media")
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/googleutils.py:120]: remove

File: backend/api/quivr_api/modules/sync/utils/googleutils.py [hunk 3]
```diff
@@ -122,28 +127,41 @@ async def _upload_files(
                     )
                     continue
 
+                logger.debug("Getting file data")
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/googleutils.py:130]: remove

File: backend/api/quivr_api/modules/sync/utils/googleutils.py [hunk 4]
```diff
@@ -122,28 +127,41 @@ async def _upload_files(
...
                 file_data = request.execute()
+                logger.debug("File data: %s", len(file_data))
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/googleutils.py:132]: remove

File: backend/api/quivr_api/modules/sync/utils/googleutils.py [hunk 5]
```diff
@@ -122,28 +127,41 @@ async def _upload_files(
...
 
                 # Check if the file already exists in the storage
                 if check_file_exists(brain_id, file_name):
                     logger.debug("🔥 File already exists in the storage: %s", file_name)
                     self.storage.remove_file(brain_id + "/" + file_name)
                     BrainsVectors().delete_file_from_brain(brain_id, file_name)
 
+                logger.debug("Uploading file")
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/googleutils.py:140]: remove

File: backend/api/quivr_api/modules/sync/utils/googleutils.py [hunk 6]
```diff
@@ -122,28 +127,41 @@ async def _upload_files(
...
                 to_upload_file = UploadFile(
                     file=BytesIO(file_data),
                     filename=file_name,
                 )
 
+                logger.debug("Checking if the file already exists in the database")
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/googleutils.py:146]: remove

File: backend/api/quivr_api/modules/sync/utils/googleutils.py [hunk 7]
```diff
@@ -122,28 +127,41 @@ async def _upload_files(
...
                 # Check if the file already exists in the database
                 existing_files = self.sync_files_repo.get_sync_files(sync_active_id)
                 existing_file = next(
                     (f for f in existing_files if f.path == file_name), None
                 )
+                logger.debug("Existing file: %s", existing_file)
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/googleutils.py:152]: remove

File: backend/api/quivr_api/modules/sync/utils/googleutils.py [hunk 8]
```diff
@@ -122,28 +127,41 @@ async def _upload_files(
...
                 supported = False
                 if (existing_file and existing_file.supported) or not existing_file:
                     supported = True
-                    await upload_file(to_upload_file, brain_id, current_user, bulk_id)  # type: ignore
+                    logger.debug("Uploading file")
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/googleutils.py:156]: remove

File: backend/api/quivr_api/modules/sync/utils/googleutils.py [hunk 9]
```diff
@@ -63,15 +65,17 @@ async def _upload_files(
...
-                file_name = file["name"]
-                mime_type = file["mime_type"]
-                modified_time = file["last_modified"]
+                file_id = file.id
+                file_name = file.name
+                mime_type = file.mime_type
+                modified_time = file.last_modified
+                file_url = file.web_view_link
                 # Convert Google Docs files to appropriate formats before downloading
+                logger.debug("🔥🔥🔥🔥: %s", mime_type)
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/googleutils.py:78]: remove

File: backend/api/quivr_api/modules/notification/dto/inputs.py [hunk 10]
```diff
@@ -16,6 +20,8 @@ class CreateNotification(BaseModel):
 
     def model_dump(self, *args, **kwargs):
         notification_dict = super().model_dump(*args, **kwargs)
+        logger.debug("Notification Dict: %s", notification_dict)
```
Comment (chloedia) [inline backend/api/quivr_api/modules/notification/dto/inputs.py:23]: remove

File: backend/api/quivr_api/modules/sync/utils/upload.py [hunk 11]
```diff
@@ -32,21 +35,25 @@ async def upload_file(
...
+    integration: Optional[str] = None,
+    integration_link: Optional[str] = None,
 ):
     validate_brain_authorization(
         brain_id, current_user, [RoleEnum.Editor, RoleEnum.Owner]
     )
     user_daily_usage = UserUsage(
         id=current_user,
     )
+    logger.debug("Uploading file in upload file")
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/upload.py:47]: remove

File: backend/api/quivr_api/modules/sync/utils/upload.py [hunk 12]
```diff
@@ -32,21 +35,25 @@ async def upload_file(
...
     upload_notification = notification_service.add_notification(
         CreateNotification(
             user_id=current_user,
-            bulk_id=bulk_id,
+            bulk_id=str(bulk_id),
             status=NotificationsStatusEnum.INFO,
             title=f"{upload_file.filename}",
         )
     )
+    logger.debug("Notification Created: %s", upload_notification)
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/upload.py:56]: remove

File: backend/api/quivr_api/modules/sync/utils/upload.py [hunk 13]
```diff
@@ -84,21 +91,30 @@ async def upload_file(
                 status_code=500, detail=f"Failed to upload file to storage. {e}"
             )
 
+    logger.debug("Creating Knowledge")
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/upload.py:94]: remove

File: backend/api/quivr_api/modules/sync/utils/upload.py [hunk 14]
```diff
@@ -84,21 +91,30 @@ async def upload_file(
...
         brain_id=brain_id,
         file_name=upload_file.filename,
         extension=os.path.splitext(
             upload_file.filename  # pyright: ignore reportPrivateUsage=none
         )[-1].lower(),
+        integration=integration,
+        integration_link=integration_link,
     )
 
+    logger.debug("Knowledge to add: %s", knowledge_to_add)
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/upload.py:105]: remove

File: backend/api/quivr_api/modules/sync/utils/upload.py [hunk 15]
```diff
@@ -84,21 +91,30 @@ async def upload_file(
...
+
     added_knowledge = knowledge_service.add_knowledge(knowledge_to_add)
+    logger.info("Added knowledge: %s", added_knowledge)
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/upload.py:108]: remove

File: backend/api/quivr_api/modules/sync/utils/upload.py [hunk 16]
```diff
@@ -84,21 +91,30 @@ async def upload_file(
...
 
+    logger.debug("Processing file and notify")
```
Comment (chloedia) [inline backend/api/quivr_api/modules/sync/utils/upload.py:110]: remove

File: backend/api/quivr_api/celery_worker.py [hunk 17]
```diff
@@ -238,7 +245,20 @@ def check_if_is_premium_user():
         supabase_db.table("product_to_features").select("*").execute()
     ).data
 
-    user_settings = (supabase_db.table("user_settings").select("*").execute()).data
+    user_settings = []
```
Comment (StanGirard) [inline backend/api/quivr_api/celery_worker.py:248]: To remove

**Pull request merged**